moved/src/types/mod.rs

```diff
@@ -1,7 +1,4 @@
 pub mod engine_api;
```

Comment:
Why don't we move this type into `engine-api` crate?

Reply:
Or alternatively we might want to separate the `moved/types` folder into its own crate named `types`, because that's where the dependency connection happens. This way `engine-api` will not depend on `moved` crate and vice versa. We will need to move `moved/primitives.rs` into the `types` crate too.